src/Microsoft.Identity.Web/MicrosoftIdentityOptions.cs

```diff
@@ -133,5 +133,34 @@ internal bool HasClientCredentials
         /// See https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal.
         /// </summary>
         public string? UserAssignedManagedIdentityClientId { get; set; }
+
+        /// <summary>
+        /// Applies to a web API: does the web API allow app-only tokens?.
+        /// </summary>
+        /// <remarks>
+        /// In the app registration you'll need to opt-in for the idtyp claim.
+        /// </remarks>
+        public bool ApiAllowsAppOnlyTokens { get; set; } = true;
```

Comment:
should not be needed for authz. If app roles are used, the role could be assigned to any security principal

Reply:
By default, Microsoft.Identity.Web verifies that a token contains at least roles or scopes, or the web API configuration claims that it's protected by ACLs.
Then the developer can add/should add the verification of roles on each controller or even controller action. But that's work.

@yordan-msft : The role could be assigned to any security principal. How does it relate to asking app-only scopes?
 @hpsin : do we want to have this option?

Reply:
This is a good "easy win" for apps that aren't able to turn on app role assignment yet.  It provides defense in depth as well.

Reply:
I disagree with app having to reason about app only or types. AuthZ is done on identifiers. If app does authZ on OIDs and groups, the principal can be anything. There is no need to exclude app only. 
Seems like we are adding this option as workaround for apps doing something wrong. If this is the case, call out what this something is.